controllers/session/session_controller.go

```diff
@@ -174,13 +174,15 @@ func (r ReconcileSession) WatchTypes() []client.Object {
 
 // Reconcile reads that state of the cluster for a Session object and makes changes based on the state read
 // and what is in the Session.Spec.
-func (r *ReconcileSession) Reconcile(c context.Context, request reconcile.Request) (reconcile.Result, error) { //nolint:cyclop,gocyclo //reason WIP
+func (r *ReconcileSession) Reconcile(orgCtx context.Context, request reconcile.Request) (reconcile.Result, error) { //nolint:cyclop,gocyclo //reason WIP
 	reqLogger := logger().WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling Session")
 
+	c := NewInstrumentedClient(r.client)
```

Comment:
One general question - do we want to collect/expose those metrics all the time? How big is overhead here?

My line of thinking is that it could potentially be enabled through a flag instead of being always on, so the user can opt-in if they wish.

Reply:
No overhead to talk about.